plugins/modules/zos_replace.py

```diff
@@ -0,0 +1,646 @@
+        module.fail_json(msg="Order of patter is incorrect, after patters was found after the before patter.")
+
+    new_text, replaced = replace_text(content=decode_list[begin_block_code:end_block_code],
+                                      regexp=regexp, replace=replace, literal=lit_rex)
+    full_new_text = merge_text(original=decode_list, replace=new_text, begin=begin_block_code, end=end_block_code)
+    return full_new_text, replaced, new_text
+
+
+def run_module():
+    module = AnsibleModule(
+        argument_spec=dict(
+            after=dict(type='str'),
+            backup=dict(type='bool', default=False, required=False),
+            backup_name=dict(type='str', default=None, required=False),
+            before=dict(type='str'),
+            encoding=dict(type='str', default='cp1047', required=False),
+            target=dict(type="str", required=True, aliases=['src', 'path', 'destfile']),
+            tmp_hlq=dict(type='str', required=False, default=None),
+            literal=dict(type="list", elements="str", required=False, default=[], options=["after", "before", "regexp"]),
+            regexp=dict(type="str", required=True),
+            replace=dict(type='str', default=""),
+        ),
+        supports_check_mode=False
+    )
+    args_def = dict(
+        after=dict(type='str'),
+        backup=dict(type='bool', default=False, required=False),
+        backup_name=dict(type='data_set_or_path', default=None, required=False),
+        before=dict(type='str'),
+        encoding=dict(type='str', default='IBM-1047', required=False),
+        target=dict(type="data_set_or_path", required=True, aliases=['src', 'path', 'destfile']),
+        tmp_hlq=dict(type='qualifier_or_empty', required=False, default=None),
+        literal=dict(type="list", elements="str", required=False, default=[], option=["after", "before", "regexp"]),
+        regexp=dict(type="str", required=True),
+        replace=dict(type='str', default=""),
+    )
+
+    try:
+        parser = better_arg_parser.BetterArgParser(args_def)
+        parsed_args = parser.parse_args(module.params)
+        module.params = parsed_args
+    except ValueError as err:
+        module.fail_json(
+            msg='Parameter verification failed.',
+            stderr=str(err)
+        )
+
+    src = module.params.get("target")
+
+    result = dict()
+    changed = False
+    result["target"] = src
+    result["changed"] = changed
+
+    tmphlq = parsed_args.get('tmp_hlq')
+    src = resolve_src_name(module=module, name=src, results=result, tmphlq=tmphlq)
+    uss = True if "/" in src else False
+    after = module.params.get("after")
+    before = module.params.get("before")
+    regexp = module.params.get("regexp")
+    replace = module.params.get("replace")
+    encoding = module.params.get("encoding")
+    backup = module.params.get("backup")
+    if parsed_args.get('backup_name') and backup:
+        backup = parsed_args.get('backup_name')
```

Comment:
We don't have a reason to reuse `backup` instead of `backup_name`, it adds complexity down the road because we don't know if this is of type `str` or `bool` and hence we need to check everytime. Please use `backup_name` for this.